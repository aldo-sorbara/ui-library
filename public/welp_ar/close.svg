<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
