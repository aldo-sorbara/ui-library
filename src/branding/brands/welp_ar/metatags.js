export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
