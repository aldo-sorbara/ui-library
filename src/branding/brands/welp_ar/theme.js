import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
