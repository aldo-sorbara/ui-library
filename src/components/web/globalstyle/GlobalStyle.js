import React from 'react';
import { createGlobalStyle } from 'styled-components';
import PropTypes from 'prop-types';

const GlobalStyle = props => {
  const { SETTINGS } = props;
  const fontFamily = `${SETTINGS.typography}:200,400,700,900`;
  const Component = createGlobalStyle`
  @import url('https://fonts.googleapis.com/css?family=${fontFamily}');
  @import url('https://necolas.github.io/normalize.css/8.0.1/normalize.css');
  * {
    box-sizing: border-box;
  }

  html,
  body {
    /* Define font-size in pixels once to set initial value. */
    font-size: ${SETTINGS.theme.font.defaultSize};
    height: 100%;
  }

  body {
    color: ${SETTINGS.palette.common.text.default};
    font-family: ${SETTINGS.typography};
    margin: 0;
    padding: 0;
  }


  input[type="number"] {
    -moz-appearance: textfield;
  }

  input[type="number"]::-webkit-outer-spin-button, input[type="number"]::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }

  a {
    color: ${SETTINGS.palette.secondary.dark};
    text-decoration: none;

    &[disabled] {
      cursor: default;
      opacity: .4;
      pointer-events: none;
    }
  }

  ul {
    margin-bottom: ${SETTINGS.theme.spacer.blockSpacer};
    margin-top: 0;
    padding: 0 1rem;

    > li {
      list-style-type: none;
      margin-bottom: .5rem;
      ${SETTINGS.theme.typography.description};
      padding-left: 1.25rem;
      position: relative;

      &:last-child {
        margin-bottom: 0;
      }

      &::before {
        background-color: ${SETTINGS.palette.common.text.default};
        border: 1px solid ${SETTINGS.palette.common.text.default};
        border-radius: 50%;
        content: "";
        display: block;
        height: .25rem;
        left: 0;
        position: absolute;
        top: .5rem;
        width: .25rem;
      }

      &:empty {
        display: none;
      }
    }
  }
  ${SETTINGS.theme.reset};
`;
  return <Component {...props} />;
};

GlobalStyle.defaultProps = {
  SETTINGS: {
    typography: 'Nunito',
    palette: {
      common: {
        text: {
          default: '#1d2c51'
        }
      },
      secondary: {
        dark: '#c346f9'
      },
      primary: {
        default: '#0082ff'
      }
    },
    theme: {
      spacer: {
        blockSpacer: '1.25rem'
      },
      typography: {
        description: {
          'font-size': '.875rem',
          'font-weight': '400',
          'line-height': '1.25'
        }
      },
      font: {
        defaultSize: '16px'
      },
      reset: {
        p: {
          'margin-top': 0,
          'margin-bottom': '.5rem'
        }
      }
    }
  }
};

GlobalStyle.propTypes = {
  SETTINGS: PropTypes.object
};

export default GlobalStyle;
