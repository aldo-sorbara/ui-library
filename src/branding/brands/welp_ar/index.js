import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
