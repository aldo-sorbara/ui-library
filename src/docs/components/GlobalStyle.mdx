---
name: GlobalStyle
menu: Components
route: /globalStyle
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import GlobalStyle from '../../components/web/globalstyle';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import { styles } from '../../../utils/docz';

# GlobalStyle

<PropsTable of={GlobalStyle} />

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <div>
      <GlobalStyle theme={welp_ar.theme} />
      <h1> h1 with GlobalStyle</h1>
      <a href='#'> link with GlobalStyle</a>
    </div>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/globalstyle--h1-with-globalstyle-and-default-props)
