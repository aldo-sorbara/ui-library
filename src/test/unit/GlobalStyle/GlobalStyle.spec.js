import React from 'react';
import GlobalStyle from '../../../components/web/globalstyle';
import renderer from 'react-test-renderer';
import welp_ar from '../../../branding/brands/welp_ar';

import 'jest-styled-components';

describe('GlobalStyle component', () => {
  it('should render globalStyle snapshot', () => {
    const component = renderer.create(<GlobalStyle theme={welp_ar.theme} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
