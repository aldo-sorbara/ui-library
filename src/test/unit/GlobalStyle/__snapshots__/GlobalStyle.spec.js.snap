// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalStyle component should render globalStyle snapshot 1`] = `null`;
